### PR TITLE
NADTelnet: Fix initial telnet open connection

### PR DESCRIPTION
**Pull request by andreas-amlabs**
NAD telnet firmwares behave differently and some
firmwares output a lot of initial data like
settings from BlueOS

### CODE REVIEW

File: nad_receiver/nad_transport.py [hunk 1]
```diff
@@ -67,17 +67,38 @@ def __init__(self, host: str, port: int, timeout: int) -> None:
...
+            # some firmwares sends nothing
+            # some firmwares sends e.g. b'\rMain.Model=T787\r\n'
+            # some firmwares sends multiple lines (BlueOS settings dump ?)
+            #    including blank lines between data lines
+
+            # To be able to handle the blank lines between the data lines
+            # force X empty lines before continuing, this will induce a slight
+            # delay of blank_lines x timeout
+            count = self.blank_lines
+            while count:
```
Comment (gladhorn) [inline nad_receiver/nad_transport.py:90]: I quite dislike the blank_lines hack, I wish we could come up with something nicer.
For me this reads that we may just needlessly delay any communication by five seconds before doing anything sensible.
Is there any way to query if there is data to be read and just read that much data?

File: nad_receiver/nad_transport.py [hunk 2]
```diff
@@ -67,17 +67,38 @@ def __init__(self, host: str, port: int, timeout: int) -> None:
...
+                try:
+                    self.telnet.read_until("\n".encode(), self.timeout)
```
Comment (gladhorn) [inline nad_receiver/nad_transport.py:92]: Could you time how long this takes? Does it reach the timeout?

File: nad_receiver/nad_transport.py (hunk 1 above)
Comment (andreas-amlabs) [reply nad_receiver/nad_transport.py:90]: Me too. Im just out of ideas and the current implementation could give back a response like

main.power=?
T787
or
main.volume=?
BlueOS

This is a real problem, will show up upon power on and "some timing" with the connection. But could cause all kinds of issues. BUT lets drop the 5 x read, keep the rest. I will upload new patch - then lets figure out a new way to handle this - I am open for ideas

File: nad_receiver/nad_transport.py (hunk 2 above)
Comment (andreas-amlabs) [reply nad_receiver/nad_transport.py]: "Does it reach the timeout", in which scenario ?
The normal case for my FW (without the x5 dummy ready) No, my
FW outputs at the very least "\rMain.Model=T787\r\n", so no timeout - for others ?
Dunno. This line is output upon every re-connect

**Comment by andreas-amlabs** [comment]
You want any changes to this PR ?

**Comment by andreas-amlabs** [comment]
Any comments on this on ? I would like to see it merged even though it does
not fix ALL issues, it does fix some of them!

**Comment by andreas-amlabs** [comment]
@joopert: Could you please merge this as the row 91 fixes a real potential issue

Thanks

File: nad_receiver/nad_transport.py (hunk 2 above)
Comment (gladhorn) [reply nad_receiver/nad_transport.py]: I meant the read_until, as in measuring the time it takes, to deduce if it actually is just waiting for the default timeout or not.

**Comment by gladhorn** [comment]
I'd like to make an attempt to find a solution without using a defined number of lines to be read. As you say, the firmware changes, and we cannot expect users to change some constant that hardly makes sense without knowing how the firmware behaves.

**Comment by gladhorn** [comment]
Ah, you updated, great, I think this looks good :)

**Pull request merged**